packages/react-components/components/PhoneNumberInput.tsx

```diff
@@ -65,6 +71,33 @@ export default class PhoneNumberInput extends React.Component<Props, State> {
     }
   }
 
+  async triggerPhoneNumberRequest() {
+    try {
+      const phone = await SmsRetriever.requestPhoneNumber()
+      const phoneNumber = parsePhoneNumberFromString(phone, undefined)
+      if (phoneNumber) {
+        this.setState({
+          phoneNumber: phoneNumber.nationalNumber.toString(),
+        })
+
+        this.onChangeCountryQuery(
+          this.state.countries.getCountryByPhoneCountryCode(
```

Comment:
How would this work given that a phone country code can be shared between countries (e.g Canada and US both +1)

Reply:
I'm not completely sure we can identify if a number is American or Canadian just with the phone number. I just looked up the number of the Toronto Tower and it says the phone number region is CA: https://libphonenumber.appspot.com/phonenumberparser?number=%2B1+416-868-6937